### PR TITLE
chore: rafiki helm chart

### LINKED ISSUE #232: Create helm chart for Rafiki
Create the initial Helm chart deployment for deploying Rafiki. Its expected that the helm chart can be used to deploy Rafiki and any dependencies it has

At a minimum we need
* Deployment
* Service for public and admin access
* Service Account/ RBAC
* Dependencies
    * Optionally Deploy Postgres
    * Optionally Deploy Tigerbeetl 

**Good examples**
[Emissary](https://github.com/emissary-ingress/emissary )under charts folder
[Cockroach](https://github.com/cockroachdb/helm-charts)

### PR DESCRIPTION
**Pull request by rluckom-coil**
## Changes proposed in this pull request
This adds a helm chart for the Rafiki aut,t, backend, and frontend services as well as a wrapper chart that shows deploying Rafiki with its databases. 

## Context

- fixes #232 

## Checklist

- [x] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [x] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

File: infrastructure/helm/rafiki/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.0.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: '0.0.0'
```
Comment (sabineschaller) [inline infrastructure/helm/rafiki/Chart.yaml:24]: This is where we should now put `1.0.0-alpha` I think.

**Review by sabineschaller — COMMENTED** [review]
Thank you!

**Comment by sabineschaller** [comment]
When I try 
```
helm install rafiki-example .
```
I get 
```
Error: INSTALLATION FAILED: error evaluating symlink /Users/sabine/git/coil/rafiki/infrastructure/local/local-http-signatures/node_modules/@interledger/http-signature-utils: lstat /Users/sabine/git/coil/rafiki/node_modules/.pnpm/@interledger+http-signature-utils@1.0.0: no such file or directory
```

**Comment by sabineschaller** [comment]
I restarted kubernetes and now it works 🤷‍♀️. But thank you for the debugging list 🙂.

File: infrastructure/helm/rafiki-example/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+### Example Rafiki deployment
+
+This chart is a wrapper around the [rafiki chart in this repo](../rafiki) and the Bitnami
+Postgres and Redis charts. The [values.yaml](./values.yaml) file shows how to configure the
+database connections for the services, assuming the chart is deployed with the name `rafiki-example`.
+
+To install this chart with [helm](https://helm.sh/docs/intro/install/) in a k8s cluster, pull this git repo
+and run `helm install rafiki-example .` from this directory.
```
Comment (sabineschaller) [inline infrastructure/helm/rafiki-example/README.md:8]:
```suggestion
and run `helm install rafiki-example .` from this directory, i.e. `./infrastructure/helm/rafiki-example`.
```

File: infrastructure/helm/rafiki-example/values.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      host: "rafiki-example-postgresql"
+      port: 5432
+      username: "backend"
+      database: "backend"
+      password: "changeme"
+    openPaymentsPort: 80
+    adminPort: 3001
+    image:
+      repository: ghcr.io/interledger/rafiki-backend
+      tag: 'd029ee4'
```
Comment (sabineschaller) [inline infrastructure/helm/rafiki-example/values.yaml:27]: Could you use `v1.0.0-alpha` here please?

File: infrastructure/helm/rafiki-example/values.yaml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    postgresql:
+      host: "rafiki-example-postgresql"
+      port: 5432
+      username: "auth"
+      database: "auth"
+      password: "changeme"
+    port: 3006
+    image:
+      repository: ghcr.io/interledger/rafiki-auth
+      tag: 'd029ee4'
```
Comment (sabineschaller) [inline infrastructure/helm/rafiki-example/values.yaml:64]: same here, `v1.0.0-alpha`

File: infrastructure/helm/rafiki/values.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    host: "postgres.example.com"
+    port: 5432
+    username: "backend"
+    database: "backend"
+    password: "overridethisValue"
+  openPaymentsPort: 80
+  adminPort: 3001
+  image:
+    repository: ghcr.io/interledger/rafiki-backend
+    tag: 'main'
```
Comment (sabineschaller) [inline infrastructure/helm/rafiki/values.yaml:23]:
```suggestion
    tag: 'v1.0.0-alpha'
```

File: infrastructure/helm/rafiki/values.yaml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  postgresql:
+    host: "postgres.example.com"
+    port: 5432
+    username: "auth"
+    database: "auth"
+    password: "overridethisValue"
+  port: 3006
+  image:
+    repository: ghcr.io/interledger/rafiki-auth
+    tag: 'main'
```
Comment (sabineschaller) [inline infrastructure/helm/rafiki/values.yaml:59]:
```suggestion
    tag: 'v1.0.0-alpha'
```

File: infrastructure/helm/rafiki/README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+### Helm Chart for Rafiki
```
Comment (sabineschaller) [inline infrastructure/helm/rafiki/README.md:1]: Maybe mention that the example values in the `values.yaml` need to be updated.

**Comment by sabineschaller** [comment]
Also, there is a merge conflict that needs to be resolved

**Review by sabineschaller — DISMISSED** [review]
In general, looks good to me. Thank you!

File: docs/deployment.md [hunk 8]
```diff
@@ -22,15 +22,79 @@ Dependencies:
...
+<!-- To add the Rafiki chart, run
 
 ```
 // add rafiki repository
 $ helm repo add ...
 
-// install rafiki components
-$ helm install ...
-```
+``` -->
```
Comment (sabineschaller) [inline docs/deployment.md:37]: I still need to figure out where we host the actual charts. This is why this is commented out. Github does to offer a chart repository. 

I'll add that information in a future PR once I've figured out where to host the chart.

**Review by BlairCurrey — DISMISSED** [review]

**Review by BlairCurrey — APPROVED** [review]

**Pull request merged**